### ISSUE TITLE
Support Microsoft Edge on Mac and Linux in flutter run

### ISSUE THREAD

**Issue by thomassth** [flutter/flutter#78153]
Continuing #55322

MS edge has stable release on Mac and currently has dev releases on Linux.
Please lift the "Windows only" restriction for `flutter run -d edge`.

In the long term, please allow user to choose and use any chromium-based browser installation for `flutter run` and not have to request support for every one of them.
Add a "at your own risk" disclaimer if you don't want to officially support them, but let the devs choose their environment.

**Comment by thomassth** [comment]
Any update on the matter?

**Comment by Quazmoz** [comment]
I would also love to have Edge and/or other chromium browsers as an option while I develop using flutter on Linux

**Comment by ddzy** [comment]
Any progress?

**Comment by LouisSung** [comment]
## What if Edge on macOS crashes immediately when started
### TL;DR
Try adding [these flags](https://github.com/flutter/flutter/blob/2.10.3/packages/flutter_tools/lib/src/web/chrome.dart#L195-L210) to start an Edge instance and see if it crashes, i.e.,
`$ /Applications/Microsoft\ Edge.app/Contents/MacOS/Microsoft\ Edge --user-data-dir='' --remote-debugging-port='54321' --disable-background-timer-throttling --disable-extensions --disable-popup-blocking --bwsi --no-first-run --no-default-browser-check --disable-default-apps --disable-translate`

### About my case
I executed `flutter run -d chrome -v` with Edge (with `export CHROME_EXECUTABLE`) and with Chrome respectively.
However, even they showed almost the same logs, the edge crashed immediately when opened (about 0.5 seconds) without any error message, and led the **`SocketException: Connection refused (OS Error: Connection refused, errno = 61)`**, as the Edge instance to connect was crashed.

It took me about an hour to find the solution, trace code, and even give up and then install the Chrome...
However, I suddenly realized, what if the crashes was not caused by Flutter, but Edge itself?
Therefore, I decided to figure out what exact the command for `-d chrome` executed to start a Chrome instance?

In my case, opening Edge with **`/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge --remote-debugging-port='54321'`**, without anything about Flutter, crashes.
Although the Edge worked normally without hanging or issues, I still tried to force quit the Edge and then everything went well now...

> **For ones who suffered from the crashes, try to run an Edge instace with flags before running into Flutter**

ps. I was using `macOS 12.2.1`, `Edge 98.0.1108.56`, and `Flutter 2.10.3`

Edge crashed:
![image](https://user-images.githubusercontent.com/37973545/159154503-9e9f17d0-34b1-4068-acfc-7466fcb4086a.png)
Chrome worked well:
![image](https://user-images.githubusercontent.com/37973545/159154507-0c61a7dc-93cc-4f28-af48-72cf286ba9fd.png)

**Comment by Chasanpro** [comment]
I am wondering still no support for edge even it is miles ahead of chrome

**Comment by LouisSung** [comment]
What do you mean no support?
I think the `export CHROME_EXECUTABLE='/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge'` works as is? 🤔

Is there any problems using such an approach?

**Comment by Chasanpro** [comment]
![image](https://user-images.githubusercontent.com/61396515/179364105-48bfc951-ea96-4fc1-bfcd-83ba2171ed9f.png)
Can anyone have any idea regarding this script 
So that we can launch FLutter web app in msedge

**Comment by Chasanpro** [comment]
> > I am wondering still no support for edge even it is miles ahead of chrome
> 
> What do you mean no support? I think the `export CHROME_EXECUTABLE='/Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge'` works as is? 🤔
> 
> Is there any problem using such an approach?

It haven't worked for me 
even chrome disappeared after running Flutter Devices command

**Comment by LouisSung** [comment]
Have you ever tried the hello world (counter) app and also got an error?
Any info like masOS, Edge, and Chrome versions? any error logs or screenshots?

**Comment by ZhuChanJuan** [comment]
oh, i encountered the same issue 
i use the counter app (which is quite sample UI), newly set up Mac Edge/Chrome/Firefox environment and Android emulators with Edge/Chrome/Firefox environment.
after trying with all methods i could found on the website such as :  [https://github.com/flutter/flutter/wiki/Running-Flutter-Driver-tests-with-Web](https://github.com/flutter/flutter/wiki/Running-Flutter-Driver-tests-with-Web)

[https://chromium.googlesource.com/external/github.com/flutter/engine/+/refs/heads/flutter-1.24-candidate.7/e2etests/web/regular_integration_tests#using-different-browsers](https://chromium.googlesource.com/external/github.com/flutter/engine/+/refs/heads/flutter-1.24-candidate.7/e2etests/web/regular_integration_tests#using-different-browsers)
I still couldn't open flutter test case on web except MacOS chrome, means the MacOS edge/firefox/safari and Android emulators chrome/edge/firefox browser all can't be triggered by flutter related commans.
So i'm wondering flutter native testing framework can only support MacOS chrome at all.

**Comment by neoacevedo** [comment]
What if Chrome or Edge are installed via flatpak?

**Comment by AhanafRASHID** [comment]
For xyz reasons, the latest edge on Ubuntu 24.04LTS seems to work fine. Just added 
`export CHROME_EXECUTABLE=/path-to-edge`
to `.bashrc` 

`Microsoft Edge 142.0.3595.65`  and 
```
Flutter 3.35.6 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 9f455d2486 (6 weeks ago) • 2025-10-08 14:55:31 -0500
Engine • hash a5f2c36e367c13f868cfe98db5806f562c52c35e (revision d2913632a4) (1
months ago) • 2025-10-07 17:26:21.000Z
Tools • Dart 3.9.2 • DevTools 2.48.0
```
